core/src/main/scala/org/http4s/util/Logging.scala

```diff
@@ -0,0 +1,118 @@
+package org.http4s.util
+
+import cats._
+import cats.implicits._
+import cats.data.Kleisli
+import cats.effect.{Effect, Sync}
+import org.http4s.Method.NoBody
+import org.http4s._
+//import org.http4s.Method._
+import org.log4s.Logger
+import org.log4s.LogLevel
+import fs2._
+
+import scala.concurrent.ExecutionContext
+
+object Logging {
+
+
+  def logMessage[F[_], A <: Message[F]](
+                                        f: Kleisli[F, A, String],
+                                        cacheBody: A => Boolean = {_: A => true},
+                                        forceBody: A => Boolean = {_: A => false},
+                                        logLevel: A => LogLevel = {_: A => org.log4s.Trace},
```

Comment:
If we're already passing in `a`, what do we get out of passing in functions of `A` for things that will usually be constant with respect to A anyway?

Reply:
The idea is that those functions allow the user to determine the functionality of the function. However while it may be constant in regard to `A` that `A` may be a `Request[F]` or a `Response[F]`  so as below this means a function that matches on _:NoBody, etc. But allows the user to pick based on criteria specified in the super - eventually typeclass.